server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadataBatch.java

```diff
@@ -0,0 +1,535 @@
+            TransportNodesListShardStoreMetadataBatch.NodeStoreFilesMetadataBatch> {
+
+    public static final String ACTION_NAME = "internal:cluster/nodes/indices/shard/store/batch";
+    public static final ActionType<TransportNodesListShardStoreMetadataBatch.NodesStoreFilesMetadataBatch> TYPE = new ActionType<>(
+        ACTION_NAME,
+        TransportNodesListShardStoreMetadataBatch.NodesStoreFilesMetadataBatch::new
+    );
+
+    private final Settings settings;
+    private final IndicesService indicesService;
+    private final NodeEnvironment nodeEnv;
+
+    @Inject
+    public TransportNodesListShardStoreMetadataBatch(
+        Settings settings,
+        ThreadPool threadPool,
+        ClusterService clusterService,
+        TransportService transportService,
+        IndicesService indicesService,
+        NodeEnvironment nodeEnv,
+        ActionFilters actionFilters
+    ) {
+        super(
+            ACTION_NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            Request::new,
+            NodeRequest::new,
+            ThreadPool.Names.FETCH_SHARD_STORE,
+            NodeStoreFilesMetadataBatch.class
+        );
+        this.settings = settings;
+        this.indicesService = indicesService;
+        this.nodeEnv = nodeEnv;
+    }
+
+    @Override
+    public void list(
+        DiscoveryNode[] nodes,
+        Map<ShardId, String> shardIdsWithCustomDataPath,
+        ActionListener<NodesStoreFilesMetadataBatch> listener
+    ) {
+        execute(new TransportNodesListShardStoreMetadataBatch.Request(shardIdsWithCustomDataPath, nodes), listener);
+    }
+
+    @Override
+    protected NodeRequest newNodeRequest(Request request) {
+        return new NodeRequest(request);
+    }
+
+    @Override
+    protected NodeStoreFilesMetadataBatch newNodeResponse(StreamInput in) throws IOException {
+        return new NodeStoreFilesMetadataBatch(in);
+    }
+
+    @Override
+    protected NodesStoreFilesMetadataBatch newResponse(
+        Request request,
+        List<NodeStoreFilesMetadataBatch> responses,
+        List<FailedNodeException> failures
+    ) {
+        return new NodesStoreFilesMetadataBatch(clusterService.getClusterName(), responses, failures);
+    }
+
+    @Override
+    protected NodeStoreFilesMetadataBatch nodeOperation(NodeRequest request) {
+        try {
+            return new NodeStoreFilesMetadataBatch(clusterService.localNode(), listStoreMetadata(request));
+        } catch (IOException e) {
+            throw new OpenSearchException("Failed to list store metadata for shards [" + request.getShardIdsWithCustomDataPath() + "]", e);
+        }
+    }
+
+    /**
+     * This method is similar to listStoreMetadata method of {@link TransportNodesListShardStoreMetadata}
+     * In this case we fetch the shard store files for batch of shards instead of one shard.
+     */
```

Comment:
Do you think it'll be helpful to provide the details about output of this method ? may be convert this comment to standard method comment which includes request, response and exceptions.

Reply:
Updated the documentation in the new commit.